doc/sphinx/proofs/writing-proofs/reasoning-inductives.rst

```diff
@@ -119,16 +119,17 @@ analysis on inductive or co-inductive objects (see :ref:`variants`).
 
 .. tacn:: destruct {+, @induction_clause } {? @induction_principle }
 
-   .. insertprodn induction_clause induction_principle
+   .. insertprodn induction_clause induction_arg
 
    .. prodn::
       induction_clause ::= @induction_arg {? as @or_and_intropattern } {? eqn : @naming_intropattern } {? @occurrences }
       induction_arg ::= @natural
       | @one_term_with_bindings
-      induction_principle ::= using @one_term {? with @bindings } {? @occurrences }
 
    Performs case analysis by generating a subgoal for each constructor of the
    inductive or co-inductive type in the :n:`@one_term` in :n:`@induction_arg`.
+   :n:`@one_term` (or the conclusion of its type--example
+   :ref:`here<example_destruct_ind_concl>`) must be of inductive or co-inductive type.
    In each new subgoal, the tactic replaces :n:`@one_term` with the associated
    constructor (applied to its arguments).   Unlike :tacn:`induction`,
    :tacn:`destruct` generates no induction hypothesis.
```

Comment:
Let's change the wording entirely to make this clearer. Something along these lines:

```suggestion
   The main argument must be a term or a premise whose type is an inductive or
   co-inductive type (or a product type whose conclusion is an inductive or
   co-inductive type). This tactic performs case analysis by generating a subgoal
   for each constructor of the inductive or co-inductive type. Unlike
   :tacn:`induction`, :tacn:`destruct` generates no induction hypothesis.
```

Then, you could insert the bullet-list explaining the various ways that the main argument can be provided.

Reply:
OK, surgery complete.  Let me know what you think.